workflow/scripts/process/process_power_2_targets.py

```diff
@@ -1,270 +1,100 @@
-            ) / (num_lsts - mask_count)
-            fix[j] = avgval
-            count_overlap += 1
-
-    assert (
-        count_overlap / llen < 0.4
-    )  # assert less than 40% is overlap (catch possible errors. None found in testing)
-
-    all = [numpy.ma.masked] * llen
-    for i in range(llen):
-        for lst in lsts:
-            if fix[i] != None:
-                all[i] = fix[i]
-            else:
-                val = lst[i]
-                if numpy.ma.is_masked(val) == False:
-                    all[i] = val
-                    break
-    assert len(all) == llen
-    return all
-
-
-def get_target_areas(box_id):
-    world_boxes_path = os.path.join(
-        output_dir, "power_processed", "all_boxes", box_id, f"geom_{box_id}.gpkg"
-    )
-    gdf = gpd.read_file(world_boxes_path)
-    box = gdf["geometry"]
 
+def get_target_areas(targets_file, box_geom):
     geod = Geod(ellps="WGS84")
+    geoms = []
+    areas_km2 = []
 
     # Targets: Binary raster showing locations predicted to be connected to distribution grid.
-    with rasterio.open(
-        os.path.join(output_dir, "input", "gridfinder", "targets.tif")
-    ) as src:
-
-        # for each country (see: code) overlay connections to distribution grid
+    with rasterio.open(targets_file) as dataset:
+        # Read the dataset's valid data mask as a ndarray.
         try:
-            data, transform = rasterio.mask.mask(src, box, crop=True)
-        except:
-            return gpd.GeoDataFrame()
-
-    geoms = []
-    centroids = []
-    areas = []
-    # Extract feature shapes and values from the array.
-    for geom, val in rasterio.features.shapes(data, transform=transform):
-        if val > 0:
-            feature = shape(geom)
-            geoms.append(feature)
-            centroids.append(feature.centroid)
-            area, perimeter = geod.geometry_area_perimeter(feature)
-            areas.append(area / 1e6)
-
-    return gpd.GeoDataFrame(
-        {"area_km2": areas, "centroid": centroids, "geometry": geoms}
+            box_dataset, box_transform = rasterio.mask.mask(dataset, [box_geom], crop=True)
+
+            # Extract feature shapes and values from the array.
+            for geom, val in rasterio.features.shapes(box_dataset, transform=box_transform):
+                if val > 0:
+                    feature = shape(geom)
+                    geoms.append(feature)
+                    area_m2, _ = geod.geometry_area_perimeter(feature)
+                    areas_km2.append(abs(area_m2 / 1e6))
+        except ValueError as ex:
+            # could be that box_geom does not overlap dataset
+            logging.info("Box may not overlap targets", ex)
+            pass
+
+    return geopandas.GeoDataFrame({
+        "area_km2": areas_km2,
+        "geometry": geoms
+    })
+
+
+def get_population(targets, population_file):
+    stats = gen_zonal_stats(
+        targets.set_crs("epsg:4326").to_crs("esri:54009").geometry,  # reprojected for raster
```

Comment:
Would be nice to read the CRS from the raster if possible. If we change the raster and forget to update this it would break.